include/boost/decimal/detail/cmath/lgamma.hpp

```diff
@@ -0,0 +1,230 @@
+
+#ifndef BOOST_DECIMAL_DETAIL_CMATH_LGAMMA_HPP
+#define BOOST_DECIMAL_DETAIL_CMATH_LGAMMA_HPP
+
+#include <array>
+#include <type_traits>
+
+#include <boost/decimal/fwd.hpp> // NOLINT(llvm-include-order)
+#include <boost/decimal/detail/type_traits.hpp>
+#include <boost/decimal/detail/concepts.hpp>
+#include <boost/decimal/numbers.hpp>
+
+namespace boost {
+namespace decimal {
+
+// Begin: Forward declarations
+template <BOOST_DECIMAL_DECIMAL_FLOATING_TYPE T>
+constexpr auto log(T x) noexcept -> std::enable_if_t<detail::is_decimal_floating_point_v<T>, T>;
+
+template <BOOST_DECIMAL_DECIMAL_FLOATING_TYPE T>
+constexpr auto sin(T val) noexcept -> std::enable_if_t<detail::is_decimal_floating_point_v<T>, T>;
+
+template <BOOST_DECIMAL_DECIMAL_FLOATING_TYPE T>
+constexpr auto sqrt(T val) noexcept -> std::enable_if_t<detail::is_decimal_floating_point_v<T>, T>;
+
+template <BOOST_DECIMAL_DECIMAL_FLOATING_TYPE T>
+constexpr auto tgamma(T x) noexcept -> std::enable_if_t<detail::is_decimal_floating_point_v<T>, T>;
+// End: Forward declarations
+
+template <BOOST_DECIMAL_DECIMAL_FLOATING_TYPE T>
+constexpr auto lgamma(T x) noexcept -> std::enable_if_t<detail::is_decimal_floating_point_v<T>, T> // NOLINT(misc-no-recursion)
+{
+    T result { };
+
+    const auto nx = static_cast<int>(x);
+
+    const auto is_pure_int = (nx == x);
+
+    const auto fpc = fpclassify(x);
+
+    if (fpc != FP_NORMAL)
+    {
+      if ((fpc == FP_ZERO) || (fpc == FP_INFINITE))
+      {
+          result = std::numeric_limits<T>::infinity();
+      }
+      else
+      {
+          result = x;
+      }
+    }
+    else if ((nx < 0) && is_pure_int)
+    {
+        // Pure negative integer argument.
+        result = std::numeric_limits<T>::infinity();
+    }
+    else
+    {
+        if (is_pure_int && ((nx == 1) || (nx == 2)))
+        {
+        }
+        else
+        {
+            constexpr T one { 1, 0 };
+
+            if (signbit(x))
+            {
+                // Reflection for negative argument.
+                const auto za = -x + one;
+
+                result = log(numbers::pi_v<T>) - log(sin(numbers::pi_v<T> * za)) - lgamma(za);
+            }
+            else
+            {
+                constexpr T half { 5, -1 };
+                constexpr T eight{ 8, 0 };
+
+                if (x < half)
+                {
+                    using coefficient_array_type = std::array<T, static_cast<std::size_t>(UINT8_C(25))>;
+
+                    #if (defined(__clang__) && (__clang__ < 6))
+                    #  pragma clang diagnostic push
+                    #  pragma clang diagnostic ignored "-Wmissing-braces"
+                    #endif
+
+                    constexpr auto coefficient_table =
+                        coefficient_array_type
+                        {
+                            // Use a Taylor series expansion of the logarithm of the gamma function.
+                            // N[Series[Log[Gamma[x]], {x, 0, 26}], 32]
```

Comment:
Thanks for including the Wolfram command. I can generate the decimal128 constants once this is merged.